packages/eslint-plugin-nx/src/rules/enforce-module-boundaries.spec.ts

```diff
@@ -355,6 +365,46 @@ describe('Enforce Module Boundaries (eslint)', () => {
       expect(failures[1].message).toEqual(message);
     });
 
+    it('should error when importing transitive npm packages', () => {
+      const failures = runRule(
+        {
+          ...depConstraints,
+          banTransitiveDependencies: true,
+        },
+        `${process.cwd()}/proj/libs/api/src/index.ts`,
+        `
+          import 'npm-package2';
+          import('npm-package2');
+        `,
+        graph
+      );
+
+      const message =
+        'Transitive dependencies are not allowed. Only packages defined in the "package.json" can be imported';
+      expect(failures.length).toEqual(2);
+      expect(failures[0].message).toEqual(message);
+      expect(failures[1].message).toEqual(message);
+    });
+
+    it('should not error when importing direct npm dependencies', () => {
+      const failures = runRule(
+        {
+          ...depConstraints,
+          allowTransitiveDependencies: false,
```

Comment:
This is not the name of the option 😄

Reply:
Thanks, it was renamed and this one sneaked through. Good catch!